account_asset_management/models/account_asset.py

```diff
@@ -0,0 +1,1092 @@
+                    seq += 1
+                    name = asset._get_depreciation_entry_name(seq)
+                    if line['date'] == last_date:
+                        # ensure that the last entry of the table always
+                        # depreciates the remaining value
+                        if asset.method in ['linear-limit', 'degr-limit']:
+                            depr_max = asset.depreciation_base \
+                                - asset.salvage_value
+                        else:
+                            depr_max = asset.depreciation_base
+                        amount = depr_max - depreciated_value
+                    else:
+                        amount = line['amount']
+                    if amount:
+                        vals = {
+                            'previous_id': depr_line.id,
+                            'amount': amount,
+                            'asset_id': asset.id,
+                            'name': name,
+                            'line_date': line['date'].strftime('%Y-%m-%d'),
+                            'init_entry': entry['init'],
+                        }
+                        depreciated_value += amount
+                        depr_line = line_obj.create(vals)
+                    else:
+                        seq -= 1
+                line_i_start = 0
+
+        return True
+
+    def _get_fy_duration(self, fy_id, option='days'):
+        """
+        Returns fiscal year duration.
+        @param option:
+        - days: duration in days
+        - months: duration in months,
+                  a started month is counted as a full month
+        - years: duration in calendar years, considering also leap years
+        """
+        fy = self.env['date.range'].browse(fy_id)
+        fy_date_start = datetime.strptime(fy.date_start, '%Y-%m-%d')
+        fy_date_stop = datetime.strptime(fy.date_end, '%Y-%m-%d')
+        days = (fy_date_stop - fy_date_start).days + 1
+        months = (fy_date_stop.year - fy_date_start.year) * 12  \
+            + (fy_date_stop.month - fy_date_start.month) + 1
+        if option == 'days':
+            return days
+        elif option == 'months':
+            return months
+        elif option == 'years':
+            year = fy_date_start.year
+            cnt = fy_date_stop.year - fy_date_start.year + 1
+            for i in range(cnt):
+                cy_days = calendar.isleap(year) and 366 or 365
+                if i == 0:  # first year
+                    if fy_date_stop.year == year:
+                        duration = (fy_date_stop - fy_date_start).days + 1
+                    else:
+                        duration = (
+                            datetime(year, 12, 31) - fy_date_start).days + 1
+                    factor = float(duration) / cy_days
+                elif i == cnt - 1:  # last year
+                    duration = (
+                        fy_date_stop - datetime(year, 01, 01)).days + 1
```

Comment:
not blocking, but `01` gives you an octal number in return - unless you're not planning to use months/days greater than 7